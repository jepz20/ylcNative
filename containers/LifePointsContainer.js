// @flow

import React from 'react'
import { View, StyleSheet, TouchableWithoutFeedback } from 'react-native'
import { PlayerDetail } from '../components'
import type { PlayerMap, ResultMap } from '../types/match'
import type { ToggleCalculatorVisibility } from '../actions/calculator'
import { Map } from 'immutable'

type Props = {
  players: PlayerMap,
  results: ResultMap,
  toggleCalculatorVisibility: ToggleCalculatorVisibility
}
const LifePointsContainer = ({
  players = Map(),
  results,
  toggleCalculatorVisibility
}: Props) => {
  const checkIndexIsEven = n => {
    return n % 2 === 0
  }
  return (
    <View style={styles.LifePointsContainerWrapper}>
      {players.valueSeq().map((player, index) => {
        const style = { borderRightWidth: checkIndexIsEven(index) ? 0 : 1 }
        return (
          <TouchableWithoutFeedback
            key={player.id}
            onPress={() => toggleCalculatorVisibility(player.id)}
          >
            <View style={{ flex: 1 }}>
              <PlayerDetail
                results={results}
                outerContainerStyle={style}
                name={player.name}
                currentPoints={player.currentPoints}
                id={player.id}
              />
            </View>
          </TouchableWithoutFeedback>
        )
      })}
    </View>
  )
}

const styles = StyleSheet.create({
  LifePointsContainerWrapper: {
    flexDirection: 'row',
    flex: 1,
    backgroundColor: 'green'
  }
})

export { LifePointsContainer }
