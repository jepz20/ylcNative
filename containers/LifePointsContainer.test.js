import React from 'react'
import { TouchableWithoutFeedback } from 'react-native'
import { Map, Record } from 'immutable'
import renderer from 'react-test-renderer'

import { LifePointsContainer } from './LifePointsContainer'

describe('LifePointsContainer', () => {
  const result = Record({ winner: null })
  const results = Map({
    '1': result({ winner: '1' }),
    '2': result({ winner: '2' }),
    '3': result({ winner: 'tie' })
  })
  const playerRE = Record({
    id: '1',
    name: 'test 1',
    currentPoints: '8000'
  })

  test('renders with no props ', () => {
    const lifePointsContainer = renderer
      .create(<LifePointsContainer />)
      .toJSON()
    expect(lifePointsContainer).toMatchSnapshot()
  })

  test('renders with 2 players', () => {
    const players = Map({
      '1': playerRE(),
      '2': playerRE({ id: '2', name: 'test 2' })
    })
    const lifePointsContainer = renderer
      .create(<LifePointsContainer results={results} players={players} />)
      .toJSON()
    expect(lifePointsContainer).toMatchSnapshot()
  })

  test('renders with 1 player', () => {
    const players = Map({
      '1': playerRE()
    })
    const lifePointsContainer = renderer
      .create(<LifePointsContainer results={results} players={players} />)
      .toJSON()
    expect(lifePointsContainer).toMatchSnapshot()
  })

  test('should toggle visibility for all players', () => {
    const players = Map({
      '1': playerRE(),
      '2': playerRE({ id: '2', name: 'test 2' })
    })
    const mockToggle = jest.fn()
    const lifePointsContainer = renderer.create(
      <LifePointsContainer
        results={results}
        players={players}
        toggleCalculatorVisibility={mockToggle}
      />
    ).root
    const touchable = lifePointsContainer.findAllByType(
      TouchableWithoutFeedback
    )
    expect(touchable.length).toBe(2)
    touchable[0].props.onPress()
    touchable[1].props.onPress()
    expect(mockToggle).toHaveBeenCalledTimes(2)
  })
})
