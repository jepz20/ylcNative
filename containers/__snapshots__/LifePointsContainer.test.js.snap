// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LifePointsContainer renders with 1 player 1`] = `
<View
  style={
    Object {
      "backgroundColor": "green",
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID={undefined}
  >
    <View
      style={
        Object {
          "backgroundColor": "#006400",
          "borderColor": "#005a00",
          "borderWidth": 1,
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 35,
            "justifyContent": "space-around",
            "marginTop": 10,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#000",
                "borderRadius": 50,
                "borderWidth": 1,
                "elevation": 2,
                "height": 30,
                "width": 30,
              },
              Object {
                "backgroundColor": "green",
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "borderColor": "#000",
                "borderRadius": 50,
                "borderWidth": 1,
                "elevation": 2,
                "height": 30,
                "width": 30,
              },
              Object {
                "backgroundColor": "red",
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "borderColor": "#000",
                "borderRadius": 50,
                "borderWidth": 1,
                "elevation": 2,
                "height": 30,
                "width": 30,
              },
              Object {
                "backgroundColor": "yellow",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 65,
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 65,
              },
            ]
          }
        >
          8000
        </Text>
      </View>
      <View
        style={
          Object {
            "height": 40,
            "justifyContent": "flex-end",
            "marginBottom": 5,
            "marginLeft": 5,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          style={
            Object {
              "borderBottomWidth": 0,
              "color": "#fff",
              "fontSize": 17,
              "fontWeight": "bold",
            }
          }
          underlineColorAndroid="transparent"
          value="test 1"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`LifePointsContainer renders with 2 players 1`] = `
<View
  style={
    Object {
      "backgroundColor": "green",
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID={undefined}
  >
    <View
      style={
        Object {
          "backgroundColor": "#006400",
          "borderColor": "#005a00",
          "borderWidth": 1,
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 35,
            "justifyContent": "space-around",
            "marginTop": 10,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#000",
                "borderRadius": 50,
                "borderWidth": 1,
                "elevation": 2,
                "height": 30,
                "width": 30,
              },
              Object {
                "backgroundColor": "green",
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "borderColor": "#000",
                "borderRadius": 50,
                "borderWidth": 1,
                "elevation": 2,
                "height": 30,
                "width": 30,
              },
              Object {
                "backgroundColor": "red",
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "borderColor": "#000",
                "borderRadius": 50,
                "borderWidth": 1,
                "elevation": 2,
                "height": 30,
                "width": 30,
              },
              Object {
                "backgroundColor": "yellow",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 65,
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 65,
              },
            ]
          }
        >
          8000
        </Text>
      </View>
      <View
        style={
          Object {
            "height": 40,
            "justifyContent": "flex-end",
            "marginBottom": 5,
            "marginLeft": 5,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          style={
            Object {
              "borderBottomWidth": 0,
              "color": "#fff",
              "fontSize": 17,
              "fontWeight": "bold",
            }
          }
          underlineColorAndroid="transparent"
          value="test 1"
        />
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID={undefined}
  >
    <View
      style={
        Object {
          "backgroundColor": "#006400",
          "borderColor": "#005a00",
          "borderWidth": 1,
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 35,
            "justifyContent": "space-around",
            "marginTop": 10,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#000",
                "borderRadius": 50,
                "borderWidth": 1,
                "elevation": 2,
                "height": 30,
                "width": 30,
              },
              Object {
                "backgroundColor": "red",
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "borderColor": "#000",
                "borderRadius": 50,
                "borderWidth": 1,
                "elevation": 2,
                "height": 30,
                "width": 30,
              },
              Object {
                "backgroundColor": "green",
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "borderColor": "#000",
                "borderRadius": 50,
                "borderWidth": 1,
                "elevation": 2,
                "height": 30,
                "width": 30,
              },
              Object {
                "backgroundColor": "yellow",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 65,
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 65,
              },
            ]
          }
        >
          8000
        </Text>
      </View>
      <View
        style={
          Object {
            "height": 40,
            "justifyContent": "flex-end",
            "marginBottom": 5,
            "marginLeft": 5,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          style={
            Object {
              "borderBottomWidth": 0,
              "color": "#fff",
              "fontSize": 17,
              "fontWeight": "bold",
            }
          }
          underlineColorAndroid="transparent"
          value="test 2"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`LifePointsContainer renders with no props  1`] = `
<View
  style={
    Object {
      "backgroundColor": "green",
      "flex": 1,
      "flexDirection": "row",
    }
  }
/>
`;
